lib/trubl/base.rb

```diff
@@ -28,7 +28,9 @@ def missing_or_exception?(response)
           nil
         end
 
-        code && (400..600).include?(code)
+        is_missing_or_exception = code && (400..600).include?(code)
```

Comment:
You could potentially abstract that method to a PORO that receives the response, then you can make a method `valid?` (or whatever you think is appropiate) that would check for errors and do the appropriate logging. This way you could encapsulate the logic related to the response that doesn't belong it the base and collection classes.

```
  def from_response(response)
      response_handler = ResponseHandler.new(response)
      return nil unless response_handler.valid?
      initialize(parse(response))
  end
```